File: src/main/java/ru/vmsystems/template/domain/model/user/UserEntity.java
package ru.vmsystems.template.domain.model.user;

import ru.vmsystems.template.domain.model.CompanyEntity;

import javax.persistence.*;

@Entity
@Table(name = "user")
public class UserEntity {
    private Long id;
    private String name;
    private String login;
    private String password;
    private boolean enabled;
    private String role;
    private CompanyEntity company;

    @Id
    @Column(name = "id")
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    @Basic
    @Column(name = "name")
    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    @Basic
    @Column(name = "login")
    public String getLogin() {
        return login;
    }

    public void setLogin(String login) {
        this.login = login;
    }

    @Basic
    @Column(name = "password")
    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    @Basic
    @Column(name = "enabled")
    public boolean isEnabled() {
        return enabled;
    }

    public void setEnabled(boolean enabled) {
        this.enabled = enabled;
    }

    @Basic
    @Column(name = "role")
    public String getRole() {
        return role;
    }

    public void setRole(String role) {
        this.role = role;
    }

    @ManyToOne
    public CompanyEntity getCompany() {
        return company;
    }

    public void setCompany(CompanyEntity company) {
        this.company = company;
    }
}

File: src/main/java/ru/vmsystems/glasscloud/user/UserEntity.kt
package ru.vmsystems.glasscloud.user

import java.util.*
import javax.persistence.Entity
import javax.persistence.Id
import javax.persistence.PrePersist
import javax.persistence.Table

@Entity
@Table(name = "user", schema = "glass")
data class UserEntity(
        @Id
        var id: UUID? = null,
        val name: String,
        val login: String,
        val password: String,
        val enabled: Boolean,
        val role: String,
        val email: String,
        val phone: String,
        val comment: String?,
        val companyId: UUID?,
        val deleted: Boolean = false
) {
    @PrePersist
    private fun prePersist() {
        id = UUID.randomUUID()
    }
}
